components/script/dom/window.rs

```diff
@@ -541,14 +529,10 @@ impl WindowMethods for Window {
             stdout.flush().unwrap();
             stderr.flush().unwrap();
         }
-
-        let (sender, receiver) = ProfiledIpc::channel(self.global().time_profiler_chan().clone()).unwrap();
-        self.send_to_constellation(ScriptMsg::Alert(s.to_string(), sender));
-
-        let should_display_alert_dialog = receiver.recv().unwrap();
-        if should_display_alert_dialog {
-            display_alert_dialog(&s);
-        }
+        let window_proxy = self.window_proxy.get().unwrap();
+        let top_level_browsing_context_id = window_proxy.top_level_browsing_context_id();
+        self.send_to_constellation(ScriptMsg::ForwardToEmbedder(EmbedderMsg::Alert(top_level_browsing_context_id,
+                                                                                   s.to_string())));
```

Comment:
Are we supposed to be blocking here? I believe you need  a sender/receiver pair, and block on `recv()`. In `ports/`, you would use the sender to notify when the user dismissed the alert.